### PR TITLE
Extraction of geth-specific methods of prover

### PR DESCRIPTION
**Pull request by Andy-Zhou2**
Extraction of geth-specific methods of prover

Core changes:

- Renaming of packages to remove naming conflicts (proof and proof/proof)
- Extracted out common interface to be used by all L2 execution layer clients, as listed at prover/state/api.go
- geth_prover package provides the wrappers around geth implementations of the interfaces

### CODE REVIEW

File: clients/geth/specular/prover/geth_prover/geth_evm_logger.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package geth_prover
+
+import "github.com/ethereum/go-ethereum/core/vm"
+
+type GethEVMLogger struct {
```
Comment (ujvl) [inline clients/geth/specular/prover/geth_prover/geth_evm_logger.go:5]: is this used anywhere

File: clients/geth/specular/prover/state/api.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	Coinbase() common.Address
+	GasLimit() uint64
+	BlockNumber() *big.Int
+	Time() *big.Int
+	Difficulty() *big.Int
+	BaseFee() *big.Int
+	Random() *common.Hash
+}
+
+type SpecularEVMLoggerInterface interface {
```
Comment (ujvl) [inline clients/geth/specular/prover/state/api.go:49]: hm is this supposed to be empty

File: clients/geth/specular/prover/geth_prover/geth_evm_logger.go (hunk 1 above)
Comment (Andy-Zhou2) [reply clients/geth/specular/prover/geth_prover/geth_evm_logger.go:5]: No, will delete that

File: clients/geth/specular/prover/state/api.go (hunk 2 above)
Comment (Andy-Zhou2) [reply clients/geth/specular/prover/state/api.go:49]: I can make L2ELClientConfig an interface instead of struct and then delete that

**Pull request merged**